File: frontend/src/schemas/aiDraft.js
import {z} from 'zod'

/** AI 出题表单 */
export const generateProblemDraftSchema = z.object({
  background: z
    .string()
    .trim()
    .min(1, '题目背景不能为空')
    .max(5000, '题目背景过长'),
  difficulty: z
    .string()
    .trim()
    .min(1, '请选择难度')
    .max(32, '难度字段过长'),
})

/** 测例脚本生成表单 */
export const generateTestScriptDraftSchema = z.object({
  problem_id: z
    .number({invalid_type_error: '题目 ID 无效'})
    .int('题目 ID 必须是整数')
    .min(1, '题目 ID 无效'),
  direction: z.string().max(5000, '生成方向过长').default(''),
  count: z
    .number({invalid_type_error: '测试点个数无效'})
    .int('测试点个数必须是整数')
    .min(1, '至少 1 组')
    .max(50, '最多 50 组')
    .default(10),
  range_info: z.string().max(2000, '数据范围描述过长').default(''),
})

/** 测例执行表单 */
export const executeTestDataDraftSchema = z.object({
  problem_id: z
    .number({invalid_type_error: '题目 ID 无效'})
    .int()
    .min(1, '题目 ID 无效'),
  code: z.string().trim().min(1, '脚本代码不能为空'),
  type: z.string().trim().min(1).max(32).default('acm'),
  language: z.string().trim().min(1).max(32).default('python'),
  source_draft_id: z.number().int().min(1).optional().nullable(),
})

File: backend/app/api/schemas/ai_draft.py
"""AI 草稿箱 API 请求体模型。"""

from typing import Any, Optional

from pydantic import BaseModel, Field


class GenerateProblemDraftBody(BaseModel):
    """提交 AI 出题异步任务。"""

    background: str = Field(min_length=1, max_length=5000)
    difficulty: str = Field(default="简单", max_length=32)


class GenerateTestScriptDraftBody(BaseModel):
    """提交测例脚本生成异步任务。"""

    problem_id: int = Field(ge=1)
    direction: str = Field(default="", max_length=5000)


class ExecuteTestDataDraftBody(BaseModel):
    """提交测例执行异步任务。"""

    problem_id: int = Field(ge=1)
    code: str = Field(min_length=1)
    type: str = Field(default="acm", max_length=32)
    language: str = Field(default="python", max_length=32)
    source_draft_id: Optional[int] = Field(default=None, ge=1)


class AskLlmBody(BaseModel):
    """同步 LLM 对话请求。"""

    system_setting: str = Field(min_length=1, max_length=10000)
    prompt: str = Field(min_length=1, max_length=50000)
    output_format: Optional[dict[str, Any]] = None


class ExecuteTestGenerationBody(BaseModel):
    """同步执行测例生成请求。"""

    problem_id: int = Field(ge=1)
    code: str = Field(min_length=1)
    type: Optional[str] = Field(default=None, max_length=32)
    language: str = Field(default="python", min_length=1, max_length=32)
